gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/DatasetDescriptor.java

```diff
@@ -16,25 +16,36 @@
  */
 package gobblin.compliance;
 
+import com.google.common.base.Optional;
+
+
 /**
  * Each Hive Dataset using gobblin-compliance for their compliance needs, must contain a dataset.descriptor property
  * in the tblproperties of a Hive Dataset.
  *
  * A dataset.descriptor is a description of a Hive dataset in the Json format.
  *
- * A compliance id is a column name in a Hive dataset to decide which records should be purged.
+ * A compliance field is a column name in a Hive dataset to decide which records should be purged.
  *
  * A dataset.descriptor must contain an identifier whose value corresponds the column name containing compliance id.
  *
  * Path to the identifier must be specified in the job properties file
  * via property dataset.descriptor.identifier.
  *
  * Example : dataset.descriptor = {Database : Repos, Owner : GitHub, ComplianceInfo : {IdentifierType : GitHubId}}
- * If IdentifierType corresponds to the identifier and GithubId is the compliance id column name, then
+ * If IdentifierType corresponds to the identifier and GithubId is the compliance field, then
  * dataset.descriptor.identifier = ComplianceInfo.IdentifierType
  *
  * @author adsharma
  */
-public interface DatasetDescriptor {
-  public String getComplianceId();
+public abstract class DatasetDescriptor {
+  protected String descriptor;
+  protected Optional<String> complianceFieldPath;
```

Comment:
Earlier it was getComplianceId() which I changed to getComplianceField() because what we are returning is the column name corresponding to complianceIds